src/app/shared/live-region/live-region.service.ts

```diff
@@ -0,0 +1,119 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+
+import { environment } from '../../../environments/environment';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class LiveRegionService {
+
+  /**
+   * The duration after which the messages disappear in milliseconds
+   * @protected
+   */
+  protected messageTimeOutDurationMs: number = environment.liveRegion.messageTimeOutDurationMs;
+
+  /**
+   * Array containing the messages that should be shown in the live region
+   * @protected
+   */
+  protected messages: string[] = [];
+
+  /**
+   * BehaviorSubject emitting the array with messages every time the array updates
+   * @protected
+   */
+  protected messages$: BehaviorSubject<string[]> = new BehaviorSubject([]);
+
+  /**
+   * Whether the live region should be visible
+   * @protected
+   */
+  protected liveRegionIsVisible: boolean = environment.liveRegion.isVisible;
+
+  /**
+   * Returns a copy of the array with the current live region messages
+   */
+  getMessages() {
+    return [...this.messages];
+  }
+
+  /**
+   * Returns the BehaviorSubject emitting the array with messages every time the array updates
+   */
+  getMessages$() {
+    return this.messages$;
+  }
+
+  /**
+   * Adds a message to the live-region messages array
+   * @param message
+   */
+  addMessage(message: string) {
+    this.messages.push(message);
+    this.emitCurrentMessages();
+
+    // Clear the message once the timeOut has passed
+    setTimeout(() => this.pop(), this.messageTimeOutDurationMs);
+  }
+
+  /**
+   * Clears the live-region messages array
+   */
+  clear() {
```

Comment:
I think this method should also clear all pending timeouts, otherwise we might incur in something like this:
- I add a message to the live region - a timeout is set for 30 seconds;
- I clear the live region after 10 seconds;
- I add another message to the live region after 15 more seconds - a new timeout is set for another 30 seconds, but the first timeout is still running;
- The newly added message is removed after only 5 seconds.

Reply:
Ah that's a situation I hadn't considered. I've added a test that covers this case and solved the issue by adding a uuid to the messages and removing them by their id.